src/wp-includes/class-wp-block.php

```diff
@@ -139,6 +139,24 @@ public function __construct( $block, $available_context = array(), $registry = n
 		$this->block_type = $registry->get_registered( $this->name );
 
 		$this->available_context = $available_context;
+		$this->update_available_context( $block, $available_context );
+
+		if ( ! empty( $block['innerHTML'] ) ) {
+			$this->inner_html = $block['innerHTML'];
+		}
+
+		if ( ! empty( $block['innerContent'] ) ) {
+			$this->inner_content = $block['innerContent'];
+		}
```

Comment:
Shouldn't this also be part of the `update_available_context()` method instead of here? Since that method accepts `$block` being passed from the outside of this class, it means the data in `$block['innerHTML']` and `$block['innerContent']` can change too.